test/runtests.jl

```diff
@@ -120,6 +121,11 @@ end
     @test_throws Exception @test_reference "references/camera.png" camera # unequal size
 end
 
+@testset "Plots as PNG images" begin
+    @test_reference "references/heatmap.png" heatmap([1 0; 0 1])
```

Comment:
```suggestion
    @test_reference "references/heatmap.png" heatmap([1 0; 0 1]) by=psnr_equality(15)
```

The default tolerance is 25(images are visually equivalent if PSNR is above 25), to use a loose test criterion and thus allow visual differences, we can set a lower number here, 15 or 10, to make it pass.

Reply:
I believe the test failures are due to the Plots backend instead of the image IO backend, perhaps use another backend could fix the test? Or I could upload a reference file from my ubuntu server tomorrow, and do an OS-dependent check here using an if branch.
How do you think?

Reply:
> I believe the test failures are due to the Plots backend instead of the image IO backend, 

That is my belief also.

> perhaps use another backend could fix the test? 

Probably. I do find PyPlot to be very dependable for consistency.
If seeting similarity lower doesn't do it then we can do that.